src/protocols/ethereum/views/TransactionDetails.vue

```diff
@@ -136,6 +137,18 @@ export default defineComponent({
     );
 
     onMounted(async () => {
+      // If the transaction is a token transfer and it is still pending use the pending transaction
+      // because API wont be synced yet
+      // pendingTransaction will have the most accurate data at this point
+      if (accountsTransactionsPending.value[transactionOwner]) {
+        const pendingTransaction: ITransaction = accountsTransactionsPending.value[transactionOwner]
+          .find((tx) => tx.hash === hash) as ITransaction;
+        if (pendingTransaction && pendingTransaction.tx.contractId !== adapter.coinContractId) {
+          transaction.value = pendingTransaction;
+          return;
+        }
+      }
```

Comment:
In my opinion this block of code is rather superfluous and I think using `return` within the `onMounted` callback is a bad practice. Consider this approach:

```suggestion
      const pendingTransaction = (accountsTransactionsPending.value[transactionOwner] || []).find((tx) => tx.hash === hash);
      if (pendingTransaction) {
        transaction.value = pendingTransaction;
      } else {
        // The logic of fetching the transaction data
```

Reply:
`if (accountsTransactionsPending.value[transactionOwner])`
This statement is used to check if the pending transactions composable is initialized. Without it, if user refreshes on a tx details page he might get stuck loading.

Reply:
Also pending transactions do not have correct fee information, thus if the transaction isn't a contract call I think it makes sense to fetch the correct data from the node